### DISCUSSION TITLE
Building from latest source code?

### DISCUSSION THREAD

**Discussion by ch00f** [photoprism/photoprism#4093] [Q&A] [answered]
I've been tracking down an issue for a few days where I can't seem to get Photoprism to transcode my videos using qsv.

According to the logs, this is the command photoprism is using:
`/usr/bin/ffmpeg -qsv_device /dev/dri/renderD128 -i input.MP4 -c:a aac -vf scale='if(gte(iw,ih), min(720, iw), -2):if(gte(iw,ih), -2, min(720, ih))',format=rgb32 -c:v h264_qsv -map 0:v:0 -map 0:a:0? -r 30 -b:v 50M -bitrate 50M -f mp4 -movflags +faststart -y output.avc`

This results in using my (very slow) CPU to render the video.  What's missing is "-hwaccel qsv" and "-hwaccel_output_format qsv"

When I run this command manually with these two flags included, the encoding speed jumps from around 15fps to 60fps (4k down to 720p encoding).

I can see that at least the hwaccel_output_format flag was added in commit [f0e25cb](https://github.com/photoprism/photoprism/commit/f0e25cb1e9adbf10cd8af7d7e9a37d6b55bb0d74).

So I can tell I'm using an older version. I tried photoprism/photoprism:develop, but I believe that was built back in November.

How can I run the latest version? or at least change the flags so that QSV works on my system?

**Comment by lastzero** [comment, marked as answer]
You can use the photoprism/photoprism:test image for this or follow the instructions in our Developer Guide if you rather want to build your own binaries from source.

**Reply by ch00f** [reply]
That works for me! Thank you! Just saved me a $500 server upgrade.

I still need to manually install libva-dev, libmfx-dev, and intel-media-va-driver-non-free to get it to work though. Is there an easy way to include that in the container on boot? Let me know if I need to open a new topic for this.

**Reply by lastzero** [reply]
Did you set the environment variables as described in our docs or the related Github issue?

**Reply by ch00f** [reply]
I have not! Looks like everything I need is in this [thread](https://github.com/photoprism/photoprism/issues/4030). Thanks!
